### ISSUE TITLE
MacOS Install Instructions Require Review & Updates

### ISSUE THREAD

**Issue by rcurty** [ccbogel/QualCoder#970]
I installed version 3.5 months ago without any issues, but the current instructions for macOS are causing multiple problems. For instance, the instructions mention installing dependencies from a requirements.txt file, but this file is missing from the distribution. Additionally, frequent warnings related to the x86 vs. M1 architecture complicate the installation process further. Other error messages have also appeared, making it difficult to follow the steps successfully.  It would be helpful if those could be tested and reviewed and the documentation updated to reflect the latest changes and address compatibility for MacOS users. Thank you!

**Comment by ccbogel** [comment]
Thanks for the information,
For now I would remain using the 3.5 version that was released last year.  When the 3.5 zip is downloaded it contains macOS install instructions in the README.md document. 

I do not have a mac to test or check how the install process works. I am reliant on others to provide the install instructions needed for macOS.
The latest code with the 3.6 version is still very much in development with the AI features needing more testing on more operating systems, such as macOS. The details of the error messages would be beneficial. As others can comment and provide advice how to resolve them. And then incorporate into the instructions to set QualCoder up on mac.

While it does not cover the most current code, I have been told that some people run the 3.5 Windows exe version within Wine on macOS.

**Comment by kaixxx** [comment]
@rcurty: Thank you for bringing this up. There was indeed an error in the instructions for macOS. In step 2, you must download the **current** code, using the green code button in the upper right and "download zip". (The old instructions were directing you to the latest released code, which is still 3.5)

I have now corrected the instructions in the readme. Could you please try again? As Colin already mentioned, we really need people that test the new version on macOS. It would be great if you could help a little with that. Thanks!

**Comment by kaixxx** [comment]
If you want to use the 3.5 release, make sure to follow the instructions in the readme.md file that comes with the download. 

Good luck with your workshop! I'm curious to know how you like the new version 3.6 A few students of mine installed it on macOS, and it seems to work fine.

**Comment by kaixxx** [comment]
Thank you very much. QualCoder needs more testing on Mac, so it's very helpful to get such feedback. The version restrictions for chromadb and chroma-hnswlib are related to an issue on Windows only, so it's probably fine to use a newer version on Mac. I will try to edit the requirements.txt accordingly. 

See you in Munich next week!

**Comment by kaixxx** [comment]
@tomasruizt Could you try this requirements.txt on macOS: https://github.com/kaixxx/QualCoder/blob/ai_finalize/requirements.txt

**Comment by tomasruizt** [comment]
> @tomasruizt Could you try this requirements.txt on macOS: https://github.com/kaixxx/QualCoder/blob/ai_finalize/requirements.txt

Installing using this `requirements.txt` file solved my installation issue

**Comment by kaixxx** [comment]
@tomasruizt: Thank you for testing! I have now included this change into the master branch: https://github.com/ccbogel/QualCoder/commit/1ee53596da26fca6d48ea8746f9bf3034495b34c

**Comment by vansorat** [comment]
Installed QualCoder Master Code today 1/28 and received error when running this script.
pip3 install -r requirements.txt

ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements.txt'

I continued with next script and was able to install and run 3.6.  
I had installed earlier in January without this error.
What is the purpose of the requirements.tx and how might it affect what I can do?  Thanks.

**Comment by kaixxx** [comment]
> What is the purpose of the requirements.tx and how might it affect what I can do? Thanks.

The requirements.txt contains a list of the required python packages to run QualCoder. Using `pip3 install -r requirements.txt` is a convenient way to install them all in one go. In your case, this wasn't necessary since you have already installed everything a few weeks ago. It is, however, always advisable to update to the latest requirements because there might be changes. 

> ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements.txt'

Make sure that you are in the root directory of the downloaded QualCoder code, or pip won't be able to find the requirements.txt file. Use `cd <QualCoder root path>` in the terminal to change to that directory, if needed.

**Comment by vansorat** [comment]
I had the folder in Downloads so I moved it to Applications.  Rather than use change directory, I opened Finder and right click on Qualcoder-master folder and selected New Terminal at folder. I then entered the scrip pip3 install -r requirements.txt and it works.  So maybe direction to change directory should go before this? Thanks!

**Comment by kaixxx** [comment]
You are right, I have changed it: https://github.com/ccbogel/QualCoder/commit/0c61a6f5a377b66e7e5f26e5e94cbd53593fe5d8
Thanks and nice to hear that it works now.

**Comment by ccbogel** [comment]
OK thanks for this.  I have updated the macOS install instructions.

**Comment by vansorat** [comment]
on a MAC OS Big Sur v 11 intel
I am not sure if I have an issue with my path
When I type python --version, 3.8 shows up (I think it was bundled with a stat program)
When I type python3 I get 3.12

Mastercode was installed but I ran into this error when launching:

Vansorat@Vansorat QualCoder-master % python3 /applications/QualCoder-master/qualcoder/__main__.py
Traceback (most recent call last):
  File "/applications/QualCoder-master/qualcoder/__main__.py", line 42, in <module>
    from PyQt6 import QtCore, QtGui, QtWidgets
ImportError: dlopen(/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/PyQt6/QtCore.abi3.so, 2): Symbol not found: _kIOMainPortDefault
  Referenced from: /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/PyQt6/Qt6/lib/QtCore.framework/Versions/A/QtCore (which was built for Mac OS X 12.0)
  Expected in: /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit

**Comment by ccbogel** [comment]
"I think pip install "numpy<2" need to read pip3 install.... with a 3 after pip"
I think this is because for you, you have multiple python versions installed, it wont be the same for everyone.

looking here for a solution: https://stackoverflow.com/questions/60761175/how-to-solve-importerror-dlopen-symbol-not-found-expected-in-flat-name

Perhaps try, with pip or pip3:
`pip uninstall pyqt6 and then pip install --no-cache-dir pyqt6`

Also, as you have python 3.12 installed, can try this, which ensure the re-install is for the python 3.12 version :
Perhaps try, with pip or pip3:

`python3.13 -m pip install --upgrade pip`
Then:
`python3.12 -m pip uninstall pyqt6 and then pip install --no-cache-dir pyqt6`

**Comment by kaixxx** [comment]
Hi all,
with today's release of version 3.6, we have also added DMG app bundles for macOS, hoping to make the setup on Mac a little easier. There is a version for the newer Macs with Apple Silicon (M1...M4-processors) and the older Intel-based ones. It would be great if somebody could test the Intel-version, since I only have access to a M1 Mac.

The current release: https://github.com/ccbogel/QualCoder/releases/tag/3.6

(There is a small issue with the visibility of the icons in the "original" style. So, I would suggest switching to the "native" style in the settings.)
